### PR TITLE
CASMPET-6820 daemonset pod to export iscsi metrics

### PR DESCRIPTION
**Pull request by aasha-hpe**
## Summary and Scope
This is  in the context of iSCSI based boot content projection feature. This has the daemonset pod to export iscsi metrics to node exporter / prometheus. 

## Issues and Related PRs
CASMPET-6793

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [issue id](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing
Tested on Odin.

### Tested on: mug
image

### CODE REVIEW

File: kubernetes/cray-sysmgmt-health/values.yaml [hunk 1]
```diff
@@ -1192,3 +1192,10 @@ smartmetrics:
     repository: artifactory.algol60.net/csm-docker/stable/quay.io/galexrt/node-exporter-smartmon
     tag: v0.1.1
     pullPolicy: IfNotPresent
+
+iscsimetrics:
+  enabled: true
```
Comment (dborman-hpe) [inline kubernetes/cray-sysmgmt-health/values.yaml:1197]: This is not used.  Should the `iscsi_daemon.yaml` file be conditional on this?

File: kubernetes/cray-sysmgmt-health/templates/iscsi/iscsi_daemon.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  template:
+    metadata:
+       name: node-exporter-iscsi
+       labels:
+         iscsi: sbps
+    spec:
+      containers:
+      - name: iscsi-container
+        image: {{ .Values.iscsimetrics.image.repository }}:{{ .Values.iscsimetrics.image.tag }}  
+        imagePullPolicy: IfNotPresent
```
[the remark this answers is not in the document]
Comment (aasha-hpe) [reply kubernetes/cray-sysmgmt-health/templates/iscsi/iscsi_daemon.yaml:22]: looks good, incorporated the same.

File: kubernetes/cray-sysmgmt-health/values.yaml (hunk 1 above)
Comment (aasha-hpe) [reply kubernetes/cray-sysmgmt-health/values.yaml:1197]: This is required for prometheus to scrape data as I understand from monitoring team.

File: kubernetes/cray-sysmgmt-health/values.yaml [hunk 3]
```diff
@@ -1192,3 +1192,10 @@ smartmetrics:
...
+  image:
+    repository: artifactory.algol60.net/csm-docker/stable/iscsi-metrics
+    tag: v1.0.0
+    pullPolicy: IfNotPresent
```
[the remark this answers is not in the document]
Comment (aasha-hpe) [reply kubernetes/cray-sysmgmt-health/values.yaml:1201]: Corrected.

File: kubernetes/cray-sysmgmt-health/templates/iscsi/iscsi_daemon.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (aasha-hpe) [reply kubernetes/cray-sysmgmt-health/templates/iscsi/iscsi_daemon.yaml]: Using if/else condition had some lint errors, so have removed if/else condition. Pls take a look.

File: kubernetes/cray-sysmgmt-health/values.yaml (hunk 1 above)
Comment (dborman-hpe) [reply kubernetes/cray-sysmgmt-health/values.yaml:1197]: Okay, thanks for the explanation

File: kubernetes/cray-sysmgmt-health/templates/iscsi/iscsi_daemon.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dborman-hpe) [reply kubernetes/cray-sysmgmt-health/templates/iscsi/iscsi_daemon.yaml]: Okay, thanks.

**Review by dborman-hpe — APPROVED** [review]
LGTM

**Review by rambabubolla — APPROVED** [review]

**Pull request merged**